rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/activity/CovidPatientRegisterActivity.java

```diff
@@ -72,7 +76,8 @@ private void languageSwitcherDialog() {
         AlertDialog.Builder builder = new AlertDialog.Builder(this);
         builder.setTitle(getResources().getString(R.string.drawer_menu_item_change_language));
         builder.setSingleChoiceItems(R.array.locales_key, selectedLanguageIndex, (dialog, position) -> {
-            LangUtils.saveLanguage(getApplication(), localesVal[position]);
+            allSharedPreferences.saveLanguagePreference(localesVal[position]);
+            Utils.updateLocale(CovidPatientRegisterActivity.this);
```

Comment:
You can use the `NativeFormLangUtils` to update the app locale on native forms.

Reply:
used `LangUtils.class`

Reply:
Does that work?

Reply:
Are the forms translated now?